src/modules/Host.rb

```diff
@@ -57,12 +57,39 @@ def main
       # "hosts" file location
       @hosts_file = "/etc/hosts"
 
-      # Only write configuration
-      @write_only = false
-
       @initialized = false
     end
 
+    # Remove all entries from the host table.
+    def clear
+      @hosts = {}
+      modified = true
```

Comment:
If we replace all `modified = true` with `@modified = true` then we can remove the only remaining`publish :variable`.
It is that way in the original patch. Thanks for porting it, it looks like I missed to merge it back then :-/